paimon-core/src/main/java/org/apache/paimon/metrics/AbstractMetricGroup.java

```diff
@@ -0,0 +1,201 @@
+     *
+     * @param name name of the histogram
+     * @param histogram histogram to register
+     * @param <H> histogram type
+     * @return the registered histogram
+     */
+    public <H extends Histogram> H histogram(String name, H histogram) {
+        Metric metric = addMetric(name, histogram);
+        return (H) metric;
+    }
+
+    /**
+     * Adds the given metric to the group and registers it at the registry, if the group is not yet
+     * closed, and if no metric with the same name has been registered before.
+     *
+     * @param metricName the name to register the metric under
+     * @param metric the metric to register
+     */
+    protected Metric addMetric(String metricName, Metric metric) {
+        if (metric == null) {
+            LOG.warn(
+                    "Ignoring attempted registration of a metric due to being null for name {}.",
+                    metricName);
+            return null;
+        }
+        // add the metric only if the group is still open
+        if (!isClosed()) {
+            switch (metric.getMetricType()) {
+                case COUNTER:
+                case GAUGE:
+                case HISTOGRAM:
+                    // immediately put without a 'contains' check to optimize the common case
+                    // (no
+                    // collision)
+                    // collisions are resolved later
+                    Metric prior = metrics.put(metricName, metric);
+
+                    // check for collisions with other metric names
+                    if (prior != null) {
+                        // we had a collision. put back the original value
+                        metrics.put(metricName, prior);
+
+                        // we warn here, rather than failing, because metrics are tools that
+                        // should not
+                        // fail the
+                        // program when used incorrectly
+                        LOG.warn(
+                                "Name collision: Group already contains a Metric with the name '"
+                                        + metricName
+                                        + "'. The new added Metric will not be reported.");
+                    }
+                    break;
+                default:
+                    LOG.warn(
+                            "Cannot add unknown metric type {}. This indicates that the paimon "
+                                    + "does not support this metric type.",
+                            metric.getClass().getName());
+            }
+        }
+        return metrics.get(metricName);
+    }
+
+    @Override
+    public Map<String, Metric> getMetrics() {
+        return Collections.unmodifiableMap(metrics);
+    }
+
+    /**
+     * Returns the name for this group, meaning what kind of entity it represents, for example
+     * "bucket".
+     *
+     * @return logical name for this group
+     */
+    public abstract String getGroupName();
```

Comment:
For example "bucket"? Bucket is just a tag for the metrics (at least for now). Shouldn't it be "compact" or "commit"?

Reply:
Thanks for your detailed reviewing, I missed this desc after re-designed the sense of group name.